crates/bevy_winit/src/lib.rs

```diff
@@ -158,6 +162,36 @@ fn change_window(world: &mut World) {
                         window.set_max_inner_size(Some(max_inner_size));
                     }
                 }
+                bevy_window::WindowCommand::SetWindowIcon { icon } => {
+                    let texture_assets = world.get_resource_mut::<Assets<Texture>>().unwrap();
+                    let window = winit_windows.get_window(id).unwrap();
+
+                    let winit_icon = match icon {
+                        Some(window_icon) => {
+                            match texture_assets.get(&window_icon.handle) {
+                                None => {
+                                    warn!("Setting window icon with texture that was not yet loaded: {}", window_icon.handle);
+                                    bevy_window.set_icon(icon); // We queue up the command again, so the icon is set as soon as possible
```

Comment:
This is an interesting strategy for a soft failure. I think I kind of like it.

Reply:
I like the thought of this however, we could potentially end up in an infinite loop here if the asset fails to load. 
Ideally we would do something like this:
check the `LoadState` of the asset, and then, if it's
1) `Loading` re-queue the asset 
2) Anything else (`NotLoaded`, `Failed`, etc...) don't re-queue the asset.

Reply:
Doesn't this also send a warning every frame that the asset is not loaded? `bevy_audio` kind of does the same with unloaded Audio handles, but there is no log statement.

Reply:
I was sure there was a way to check for the LoadState, but did not look deep enough. Thanks for the hint :)
Whether we should issue a warn! log:
Looking at it now it does not appear to be a very useful warning. If the asset actually fails to load we already have a error message.

When the example actually works, I will try it out and probably remove it.